objects/client.c

```diff
@@ -1985,27 +1986,34 @@ client_unmanage(client_t *c, bool window_valid)
  * \param c The client to kill.
  */
 void
-client_kill(client_t *c)
+client_kill(client_t *c, int sig)
 {
-    if(client_hasproto(c, WM_DELETE_WINDOW))
+    if (sig == -1)
     {
-        xcb_client_message_event_t ev;
+        if(client_hasproto(c, WM_DELETE_WINDOW))
+        {
+            xcb_client_message_event_t ev;
 
-        /* Initialize all of event's fields first */
-        p_clear(&ev, 1);
+            /* Initialize all of event's fields first */
+            p_clear(&ev, 1);
 
-        ev.response_type = XCB_CLIENT_MESSAGE;
-        ev.window = c->window;
-        ev.format = 32;
-        ev.data.data32[1] = globalconf.timestamp;
-        ev.type = WM_PROTOCOLS;
-        ev.data.data32[0] = WM_DELETE_WINDOW;
+            ev.response_type = XCB_CLIENT_MESSAGE;
+            ev.window = c->window;
+            ev.format = 32;
+            ev.data.data32[1] = globalconf.timestamp;
+            ev.type = WM_PROTOCOLS;
+            ev.data.data32[0] = WM_DELETE_WINDOW;
 
-        xcb_send_event(globalconf.connection, false, c->window,
-                       XCB_EVENT_MASK_NO_EVENT, (char *) &ev);
+            xcb_send_event(globalconf.connection, false, c->window,
+                        XCB_EVENT_MASK_NO_EVENT, (char *) &ev);
+        }
+        else
+            xcb_kill_client(globalconf.connection, c->window);
     }
     else
-        xcb_kill_client(globalconf.connection, c->window);
+    {
+        kill(c->pid, sig);
```

Comment:
Better idea: Export a wrapper around `kill` and another around `gethostname` to Lua, then this functionality can be implemented in Lua and someone will eventually find other uses for this.